File: docs/snippets/all/archetypes/series_lines_style.py
"""Log a scalar over time."""

from math import cos, sin, tau

import rerun as rr

rr.init("rerun_example_series_line_style", spawn=True)

# Set up plot styling:
# They are logged as static as they don't change over time and apply to all timelines.
# Log two lines series under a shared root so that they show in the same plot by default.
rr.log("trig/sin", rr.SeriesLines(colors=[255, 0, 0], names="sin(0.01t)", widths=2), static=True)
rr.log("trig/cos", rr.SeriesLines(colors=[0, 255, 0], names="cos(0.01t)", widths=4), static=True)

# Log the data on a timeline called "step".
for t in range(int(tau * 2 * 100.0)):
    rr.set_time("step", sequence=t)

    rr.log("trig/sin", rr.Scalars(sin(float(t) / 100.0)))
    rr.log("trig/cos", rr.Scalars(cos(float(t) / 100.0)))

File: docs/snippets/all/archetypes/series_lines_style.cpp
// Log a scalar over time.

#include <rerun.hpp>

#include <cmath>

constexpr float TAU = 6.28318530717958647692528676655900577f;

int main(int argc, char* argv[]) {
    const auto rec = rerun::RecordingStream("rerun_example_series_line_style");
    rec.spawn().exit_on_failure();

    // Set up plot styling:
    // They are logged static as they don't change over time and apply to all timelines.
    // Log two lines series under a shared root so that they show in the same plot by default.
    rec.log_static(
        "trig/sin",
        rerun::SeriesLines()
            .with_colors(rerun::Rgba32{255, 0, 0})
            .with_names("sin(0.01t)")
            .with_widths(2.0f)
    );
    rec.log_static(
        "trig/cos",
        rerun::SeriesLines()
            .with_colors(rerun::Rgba32{0, 255, 0})
            .with_names("cos(0.01t)")
            .with_widths(4.0f)
    );

    // Log the data on a timeline called "step".
    for (int t = 0; t < static_cast<int>(TAU * 2.0 * 100.0); ++t) {
        rec.set_time_sequence("step", t);

        rec.log("trig/sin", rerun::Scalars(sin(static_cast<double>(t) / 100.0)));
        rec.log("trig/cos", rerun::Scalars(cos(static_cast<double>(t) / 100.0)));
    }
}
